Maintainer: Artjom Slepnjov "Shellgen"
Description: Ignore a cannot create netlink socket: Permission denied
Ver: dnsmasq-2.91 or newer version.
Date: 2026-09-30 08:00 UTC
==============================================================
--- a/src/netlink.c
+++ b/src/netlink.c
@@ -83,7 +83,12 @@ static void setup_netlink(void)
   
   if (daemon->netlinkfd == -1 ||
       getsockname(daemon->netlinkfd, (struct sockaddr *)&addr, &slen) == -1)
-    die(_("cannot create netlink socket: %s"), NULL, EC_MISC);
+  {
+    if (errno != EPERM && errno != EACCES)
+      die(_("cannot create netlink socket: %s"), NULL, EC_MISC);
+    syslog(LOG_WARNING, _("netlink socket unavailable (EPERM), interface monitoring disabled"));
+    daemon->netlinkfd = -1;
+  }
   
   /* save pid assigned by bind() and retrieved by getsockname() */
   netlink_pid = addr.nl_pid;
